### PR TITLE
fix metric reporting

### PR DESCRIPTION
**Pull request by fr0stbyte**
ran into the same issue as #637 with partitions with 0 offsets - the patch below worked for me

### CODE REVIEW

**Comment by arthurvanduynhoven** [comment]
@bai Any update on ETA for this MR to be accepted/merged?

File: core/internal/httpserver/prometheus.go [hunk 1]
```diff
@@ -77,19 +76,18 @@ func (hc *Coordinator) handlePrometheusMetrics() http.HandlerFunc {
...
 						"cluster":        cluster,
 						"consumer_group": consumer,
 						"topic":          partition.Topic,
 						"partition":      strconv.FormatInt(int64(partition.Partition), 10),
 					}
 
-					consumerPartitionCurrentOffset.With(labels).Set(float64(partition.End.Offset))
 					consumerPartitionLagGauge.With(labels).Set(float64(partition.CurrentLag))
+
+					if partition.Complete >= 1.0 {
```
Comment (d1egoaz) [inline core/internal/httpserver/prometheus.go:88]: could `Complete` be greather than 1, or is this only a percentage and we could use here `== 1.0`?

File: core/internal/httpserver/prometheus_test.go [hunk 2]
```diff
@@ -151,6 +151,5 @@ func TestHttpServer_handlePrometheusMetrics(t *testing.T) {
 	assert.Contains(t, promExp, `burrow_kafka_topic_partition_offset{cluster="testcluster",partition="1",topic="testtopic"} 5566`)
 	assert.Contains(t, promExp, `burrow_kafka_topic_partition_offset{cluster="testcluster",partition="0",topic="testtopic1"} 54`)
```
Comment (d1egoaz) [inline core/internal/httpserver/prometheus_test.go:152]: wasn't the idea to get a lag report for the topic `incomplete`? I think we are missing and assertion above as  `consumerPartitionLagGauge.With(labels).Set(float64(partition.CurrentLag))` will be called now even if `Complete` is below 1

**Review by d1egoaz — CHANGES_REQUESTED** [review]
@fr0stbyte it looks like the why of these can only be found in https://github.com/linkedin/Burrow/issues/637#issuecomment-641517141, I'd rather have this information as part of the commit messages of this PR.  

there is also some new assertion to be made on the tests to check that this PR is doing what is supposed to be doing

**Comment by mwain** [comment]
I came across the same issue earlier this week, sorry i wasn't keeping an eye on the repo
Fix im using is https://github.com/linkedin/Burrow/compare/linkedin:master...mwain:fix-prom-incomplete 

Which is very similar to this one 👍

**Comment by omaskery** [comment]
Hullo, is there anything I can do to help move this PR forward? :)

**Comment by fr0stbyte** [comment]
Sorry folks, I have been busy at work, will try to take another pass at this in the following days.

**Comment by Fran-ELS-AMS** [comment]
I have the same issue and would love to see this PR merged.

Has there been any progress?

**Comment by fr0stbyte** [comment]
@d1egoaz let me know if the tests added are sufficient or you have other comments

**Comment by CurtisEppel** [comment]
Going on a full year now - how can I help get this merged? 

Tagging @bai as a code owner.

**Comment by CurtisEppel** [comment]
Thanks all! 🎉 🚢

**Pull request merged**